tests/integration/test_materialized_mysql_database/materialized_with_ddl.py

```diff
@@ -2586,6 +2586,20 @@ def named_collections(clickhouse_node, mysql_node, service_name):
         f"/* expect: (1, 'a', 1), (2, 'b', 2) */ SELECT * FROM {db}.t1",
         "1\ta\t1\n2\tb\t2\n",
     )
+    clickhouse_node.query(f"ALTER NAMED COLLECTION {db} SET port=9999")
+    clickhouse_node.query(f"DETACH DATABASE {db}")
+    mysql_node.query(f"INSERT INTO {db}.t1 VALUES (3, 'c', 3)")
+    with pytest.raises(QueryRuntimeException) as exc:
+        clickhouse_node.query(f"ATTACH DATABASE {db}")
+    assert "ConnectionFailed:" in str(exc.value)
```

Comment:
just a note: you can also use `clickhouse_node.query_and_get_error(...)`, it will return an error message if it happened during query, and if an error did not happen - it will throw exception "expected an error, but query succeeded"